packages/ckeditor5-dev-env/lib/translations/createpotfiles.js

```diff
@@ -135,12 +141,14 @@ function assertNoMissingContext( { packageContexts, sourceMessages, logger } ) {
 
 	for ( const sourceMessage of sourceMessages ) {
 		if ( !contextIdOrigins.has( sourceMessage.id ) ) {
-			logger.error(
-				`Context for the message id is missing ('${ sourceMessage.id }' from ${ sourceMessage.filePath }).`
-			);
+			errors.push( `Context for the message id is missing ('${ sourceMessage.id }' from ${ sourceMessage.filePath }).` );
 		}
 	}
 
+	for ( const error of errors ) {
```

Comment:
This part is repeated in all three functions. Wouldn't it be better to do this logic outside of `assert*` functions once?

Instead of

```
const foundErrors = [
		assertNoMissingContext( { packageContexts, sourceMessages, logger } ),
		assertAllContextUsed( { packageContexts, sourceMessages, logger } ),
		assertNoRepeatedContext( { packageContexts, logger } )
	].filter( errors => errors.length ).length > 0;
```

one could do:

```
const errors = [].concat(
		assertNoMissingContext( { packageContexts, sourceMessages, logger } ),
		assertAllContextUsed( { packageContexts, sourceMessages, logger } ),
		assertNoRepeatedContext( { packageContexts, logger } )
	);

// (...)

for ( const error of errors ) {
	logger.error( error );

	process.exitCode = 1;
}
```

Reply:
Done in 03983fb.